src/view/compareChangesTreeDataProvider.ts

```diff
@@ -21,91 +22,129 @@ export class CompareChangesTreeProvider implements vscode.TreeDataProvider<TreeN
+		// plus the current branch which may not be published yet. Check the
+		// status of the current branch using local git, otherwise assume it is from
+		// GitHub.
+		if (this.repository.state.HEAD?.name === branch) {
+			const compareBranch = await this.repository.getBranch(branch);
+			this.compareHasUpstream = !!compareBranch.upstream;
+		} else {
+			this.compareHasUpstream = true;
+		}
+	}
+
+	async updateCompareBranch(branch: string): Promise<void> {
+		await this.updateHasUpstream(branch);
+		this.compareBranchName = branch;
+		this._onDidChangeTreeData.fire();
+	}
+
+	get compareOwner(): string {
+		return this._compareOwner;
+	}
+
+	set compareOwner(owner: string) {
+		this._gitHubRepository = this.folderRepoManager.gitHubRepositories.find(
+			repo => repo.remote.owner === owner,
+		);
+
+		if (this._contentProvider && this._gitHubRepository) {
+			this._contentProvider.gitHubRepository = this._gitHubRepository;
+		}
+
+		this._compareOwner = owner;
+		this._onDidChangeTreeData.fire();
+	}
+
 	getTreeItem(element: TreeNode): vscode.TreeItem | Thenable<vscode.TreeItem> {
 		return element.getTreeItem();
 	}
 
 	async getChildren() {
 		// If no upstream, show error.
-		if (!this.compareBranch?.upstream) {
+		if (!this.compareHasUpstream) {
 			vscode.commands.executeCommand('setContext', 'github:noUpstream', true);
 			this._view.message = undefined;
 			return [];
 		} else {
 			vscode.commands.executeCommand('setContext', 'github:noUpstream', false);
 		}
 
-		const upstream = this.compareBranch.upstream!.remote;
+		if (!this._gitHubRepository) {
+			return [];
+		}
 
 		if (!this._contentProvider) {
-			this._contentProvider = new GitHubContentProvider(this.folderRepoManager, upstream);
+			this._contentProvider = new GitHubContentProvider(this._gitHubRepository);
 			this._disposables.push(
 				vscode.workspace.registerTextDocumentContentProvider('github', this._contentProvider),
 			);
 		}
 
-		const githubRepository = this.folderRepoManager.gitHubRepositories.find(
-			repo => repo.remote.remoteName === upstream,
-		);
-		if (!githubRepository) {
+		const { octokit, remote } = await this._gitHubRepository.ensure();
+
+		try {
+			const { data } = await octokit.repos.compareCommits({
+				repo: remote.repositoryName,
+				owner: remote.owner,
+				base: `${this.baseOwner}:${this.baseBranchName}`,
+				head: `${this.compareOwner}:${this.compareBranchName}`,
+			});
+
+			if (!data.files.length) {
+				this._view.message = `The are no commits between the base '${this.baseBranchName}' branch and the comparing '${this.compareBranchName}' branch`;
+			} else {
+				this._view.message = undefined;
+			}
+
+			return data.files.map(file => {
+				return new GitHubFileChangeNode(
+					this._view,
+					file.filename,
+					file.previous_filename,
+					getGitChangeType(file.status),
+					data.merge_base_commit.sha,
+					this.compareBranchName,
+				);
+			});
+		} catch (e) {
```

Comment:
Is it expected that this ever happen, or does something need to go seriously wrong to hit this catch? If the former, it might be helpful to include a notification about how it can be fixed.

Reply:
This isn't expected to happen, there's not really an action to take to fix it. Changing to use `Logger`, though